src/os/macosx/string_osx.cpp

```diff
@@ -306,6 +306,9 @@ int MacOSStringCompare(const char *s1, const char *s2)
 	CFStringRef cf1 = CFStringCreateWithCString(kCFAllocatorDefault, s1, kCFStringEncodingUTF8);
 	CFStringRef cf2 = CFStringCreateWithCString(kCFAllocatorDefault, s2, kCFStringEncodingUTF8);
 
+	/* If any CFString could not be created (e.g., due to UTF8 invalid chars), return OS unsupported functionality */
+	if (cf1 == nullptr || cf2 == nullptr) return 0;
```

Comment:
Unfortunately, this if has to get a bit longer. If just one string is `nullptr`, you'd still have to `CFRelease` the other one. For some obscure reason, `CFRelease` crashes on `nullptr`, which means nested if's to see what the release.

No release means memory leak, and if one of the strings is e.g. a town or station name, the comparer could be called many times.

Reply:
Oh! That's completely right. I don't know how I didn't even realise about that!

Reply:
Done